installer/Controller/Upgrader/Traits/UpgraderUtilityTrait.php

```diff
@@ -19,47 +19,50 @@
 
 namespace OrangeHRM\Installer\Controller\Upgrader\Traits;
 
-use mysqli;
+use Doctrine\DBAL\Schema\AbstractSchemaManager;
+use Exception;
+use OrangeHRM\Installer\Util\Connection;
 
 trait UpgraderUtilityTrait
 {
-    private $dbConnection = null;
-
-    /***
-     * @param string $host
-     * @param string $username
-     * @param string $password
-     * @param string $dbname
-     * @param int|null $port
+    /**
      * @return bool
      */
-    public function checkDatabaseConnection(
-        string $host,
-        string $username,
-        string $password,
-        string $dbname,
-        ?int $port
-    ): bool {
-        $this->dbConnection = @new mysqli($host, $username, $password, $dbname, $port);
-        return !$this->dbConnection->connect_error;
+    public function checkDatabaseConnection(): bool
+    {
+        try {
+            $connection = $this->getConnection();
+            $connection->connect();
+            return true;
+        } catch (Exception $exception) {
```

Comment:
Log the exception,
```php
use OrangeHRM\Installer\Util\Logger;
// ...
Logger::getLogger()->error($e->getMessage());
Logger::getLogger()->error($e->getTraceAsString());
```
https://github.com/RajithaKumara/orangehrm/blob/OHRM5X-1254/installer/Controller/AbstractInstallerRestController.php#L78-L79

Reply:
done